utils/run-topologically/profiler.js

```diff
@@ -0,0 +1,86 @@
+"use strict";
+
+const fs = require("fs-extra");
+const upath = require("upath");
+
+const hrtimeToMicroseconds = hrtime => {
+  return (hrtime[0] * 1e9 + hrtime[1]) / 1000;
+};
+
+const range = len => {
+  return Array(len)
+    .fill()
+    .map((_, idx) => idx);
+};
+
+const generateOutputname = () => {
+  const now = new Date(); // 2011-10-05T14:48:00.000Z
+  const datetime = now.toISOString().split(".")[0]; // 2011-10-05T14:48:00
+  const datetimeNormalized = datetime.replace(/-|:/g, ""); // 20111005T144800
+  return `Lerna-Profile-${datetimeNormalized}.json`;
+};
+
+class Profiler {
```

Comment:
This really feels like a separate, general purpose module completely independent of Lerna. I'm also surprised it doesn't exist already, accepting a callback and output options, et cetera.

Reply:
I couldn't find any. If you have any suggestions, I'm totally open to use one. Having said that, I do feel that the part that could be generalized is pretty small compared to the parts that are specific for our use case.

In any case, I'm happy to create a separate profiler package in Lerna.